provider/provider_test.go

```diff
@@ -4,134 +4,56 @@ package gcp
 
 import (
 	"context"
-	"fmt"
-	"path/filepath"
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 
 	"github.com/pulumi/providertest"
-	"github.com/pulumi/providertest/flags"
 	"github.com/pulumi/pulumi-gcp/provider/v7/pkg/version"
 	pfbridge "github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
 	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 )
 
-func TestDNSRecordSet(t *testing.T) {
-	runTest(t, test(t, "test-programs/dns-recordset"))
-}
-
-func TestPubSubSubscription(t *testing.T) {
-	runTest(t, test(t, "test-programs/pubsub-subscription"))
-}
-
-func TestPubSubTopic(t *testing.T) {
-	runTest(t, test(t, "test-programs/pubsub-topic"))
-}
-
-func TestServiceAccount(t *testing.T) {
-	runTest(t, test(t, "test-programs/serviceaccount-account",
-		providertest.WithSkippedUpgradeTestMode(providertest.UpgradeTestMode_PreviewOnly,
-			"TODO[pulumi/providertest#7] PreviewOnly is confused about stack names"),
-	))
-}
-
-func TestStorageBucket(t *testing.T) {
-	runTest(t, test(t, "test-programs/storage-bucket"))
-}
-
-func TestStorageBucketObject(t *testing.T) {
-	t.Skipf("TODO[pulumi/providertest#2] skipping because Assets are not working yet")
-	runTest(t, test(t, "test-programs/storage-bucketobject"))
-}
-
-func TestSecretManagerSecret(t *testing.T) {
-	runTest(t, test(t, "test-programs/secretmanager-secret"))
-}
-
-func TestSqlUser(t *testing.T) {
-	runTest(t, test(t, "test-programs/sql-user"))
-}
-
-func TestBigQueryTable(t *testing.T) {
-	runTest(t, test(t, "test-programs/bigquery-table",
-		providertest.WithSkippedUpgradeTestMode(providertest.UpgradeTestMode_PreviewOnly,
-			"TODO[pulumi/providertest#7] PreviewOnly is confused about stack names"),
-	))
-}
-
-func TestComputeFirewall(t *testing.T) {
-	runTest(t, test(t, "test-programs/compute-firewall"))
-}
-
-func TestCloudFunction(t *testing.T) {
-	t.Skipf("TODO[pulumi/providertest#2] skipping because Assets are not working yet")
-	runTest(t, test(t, "test-programs/cloudfunctions-function"))
+func TestUpgradeCoverage(t *testing.T) {
+	providertest.ReportUpgradeCoverage(t)
 }
 
-// Test programs that were automatically extracted from examples without autocorrection.
-func TestAutoExtractedPrograms(t *testing.T) {
-	type testCase struct {
-		program string
+func test(t *testing.T, dir string, opts ...providertest.Option) *providertest.ProviderTest {
+	envVars := map[string]string{
+		"gcp:config:project": "GOOGLE_PROJECT",
+		"gcp:config:zone":    "GOOGLE_ZONE",
+		"gcp:config:region":  "GOOGLE_REGION",
 	}
 
-	testCases := []testCase{
-		{"monitoring-alertpolicy-1"},
-		{"bigquery-datasetaccess-3"},
-		{"bigquery-routine-1"},
-		{"bigquery-routine-2"},
-		{"cloudrunv2-job-6"},
-		{"cloudrunv2-service-4"},
-		{"cloudrunv2-service-5"},
-		{"cloudrunv2-service-7"},
-		{"compute-disk-1"},
-		{"compute-disk-3"},
+	defaultValues := map[string]string{
+		"gcp:config:project": "pulumi-development",
```

Comment:
This is available via env var as well: `GCP_PROJECT_ID`.

Reply:
This is certainly skewing to Pulumi-internal setup but I find it convenient.